zubhub_frontend/zubhub/src/components/avatar_button/AvatarButton.jsx

```diff
@@ -23,25 +23,35 @@ const AvatarButton = props => {
     }
   };
 
-  const onClickSignUp = () =>{
-    (!auth.token ? setDropdownOpen(!dropdownOpen) : goToProfile())
-    setsignupStyles(true)
-  } 
+  const onClickSignUp = () => {
+    !auth.token ? setDropdownOpen(!dropdownOpen) : goToProfile();
+    setsignupStyles(true);
+  };
 
-  const oncloseSignUp = () =>{
-    setDropdownOpen(false)
-    setsignupStyles(false)
-  }
+  const oncloseSignUp = () => {
+    setDropdownOpen(false);
+    setsignupStyles(false);
+  };
   return (
     <ClickAwayListener onClickAway={oncloseSignUp}>
       <div>
-        <Avatar
-          ref={buttonRef}
-          onClick={onClickSignUp}
-          className={commonClasses.iconBox}
-          alt={auth?.username?.toUpperCase()}
-          src={auth?.avatar}
-        />
+        {!auth.token ? (
+          <Avatar
```

Comment:
this looks same may be just change props if token is available

Reply:
Thank you for the review, I have refactored the code.